### PR TITLE
Add new no-auth route for MSTeams

### PR DESCRIPTION
**Pull request by CalvinAllen**
Adds a new route to generate (essentially) a login token, but for MSTeams connections. Need it to be returned through response and not emailed since it will be displayed inside of CS. Added an override to allow specifying the expiresAt during the insertion of a new token, because I really only want to "update" the token by adding a new prop and removing an old one (and wanted to keep the original expiresAt)

[Changes reviewed on CodeStream](https://codestream-stg.staging-service.newrelic.com/r/Ya5We-trHA5di0uy/E_nxuWj0Qra1Fx8k9Lw9-w?src=GitHub) by calvinallen on Apr 6, 2023

<sup> Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

### CODE REVIEW

File: api_server/modules/users/generate_msteams_connect_code.js [hunk 1]
```diff
@@ -28,34 +28,35 @@ class GenerateMSTeamsConnectCodeRequest extends RestfulRequest {
...
-            target: "msteams"
+	async updateUserSignupToken () {
+		const { teamIds } = this.request.body;
+
+		const signupTokens = new SignupTokens({ api: this.api });
+		signupTokens.initialize();
+		// replace the hyphens with nothing as it makes copy/pasting easier
+		const tenantToken = UUID().replace(/-/g, '');
+		await signupTokens.insert(tenantToken, this.user.id, {
+			expiresIn: 600000,
```
Comment (cstryker) [inline api_server/modules/users/generate_msteams_connect_code.js:35]: So actually, a signup tokens "service" (and by "service" I really mean an internally maintained singleton object that encapsulates certain functionality, and not an actual service) is initialized once at startup time, and it is probably better to use that then instantiate a new one here ... though the functionality might ultimately end up the same, this way of doing things would be less fragile against future code changes. Grep the source for 'signupTokens.insert(' to see several examples. So better to use: `this.api.services.signupTokens.insert(...)` here. I looked at the code and saw that SigninFlowUtils.insertToken uses the new instantiation pattern, but this is something I failed to detect at the time, probably not worth updating that code (or is it even relevant anymore), but let's do things the right way here. Make sense?

Comment (CalvinAllen) [reply api_server/modules/users/generate_msteams_connect_code.js:35]: Okay, gotcha - thats actually where I stole that from

**Comment by CalvinAllen** [comment]
@cstryker Take another look, please!  I realized a couple things after our discussion -

1. This can be behind auth and reference the current user, and doesn't need any extra request body params
2. The tenantId that we were storing on the signup token is pointless because we never referenced it again - even in the complete logic of the teams database adapter

**Review by cstryker — APPROVED** [review]

**Comment by cstryker** [comment]
@CalvinAllen was this merged? should we close this?

**Comment by CalvinAllen** [comment]
@cstryker No, not yet - pending in a couple weeks last I talked with Brian

**Pull request merged**